### ISSUE TITLE
ObjectMapper configure does not accept JsonReadFeature enum

### ISSUE THREAD

**Issue by theGirrafish** [FasterXML/jackson-core#626]
* Java version: 8
* Jackson version: 2.11.0

I'm trying to configure my `ObjectMapper` to accept `NaN` values. The way to do this before version 2.10.0 was:

```java
import com.fasterxml.jackson.core.JsonParser.Feature;
import com.fasterxml.jackson.databind.ObjectMapper;

ObjectMapper mapper = new ObjectMapper();
mapper.configure(JsonParser.Feature.ALLOW_NON_NUMERIC_NUMBERS, true);
```

After version 2.10.0, `JsonParser.Feature.ALLOW_NON_NUMERIC_NUMBERS` was put on deprecation in favor of `JsonReadFeature.ALLOW_NON_NUMERIC_NUMBERS`, although when trying to configure the `ObjectMapper`, it throws a compile error since it's not the proper type.

```java
import com.fasterxml.jackson.core.json.JsonReadFeature;
import com.fasterxml.jackson.databind.ObjectMapper;

ObjectMapper mapper = new ObjectMapper();
mapper.configure(JsonReadFeature.ALLOW_NON_NUMERIC_NUMBERS, true);
```

Error:
```
The method configure(MapperFeature, boolean) in the type ObjectMapper is not applicable for the arguments (JsonReadFeature, boolean)
```

Deprecation warning for `JsonParser.Feature.ALLOW_NON_NUMERIC_NUMBERS`:
```
Deprecated Since 2.10 use com.fasterxml.jackson.core.json.JsonReadFeature.ALLOW_NON_NUMERIC_NUMBERS instead
```

**Comment by cowtowncoder** [comment]
Correct, you will either need to use deprecated features (which will still work for 2.x), or use the new builder-style construction:

```
ObjectMapper mapper = JsonMapper.builder()
    .enable(JsonReadFeature.ALLOW_NON_NUMERIC_NUMBERS)
    .build();
```

**Comment by theGirrafish** [comment]
Ahh sounds good! Is there a timeline set for the 3.x release?

https://github.com/FasterXML/jackson/wiki/Jackson-Release-3.0

**Comment by cowtowncoder** [comment]
@theGirrafish No, unfortunately at this point it is still quite ways off. Builder-style is available since 2.10, however, and I am trying to add forward-compatibility features that make sense within 2.x.
